packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -165,8 +165,8 @@ export abstract class BucketRef extends cdk.Construct {
    *      bucket is returned.
    * @returns an ObjectS3Url token
    */
-  public urlForObject(key?: any): string {
-    const components = [ 'https://', 's3.', new cdk.AwsRegion(), '.', new cdk.AwsURLSuffix(), '/', this.bucketName ];
+  public urlForObject(key?: string): string {
+    const components = [ 'https://', 's3.', cdk.Aws.region, '.', cdk.Aws.urlSuffix, '/', this.bucketName ];
```

Comment:
Would merge the `https://` and `s3.` literals... There is no point computing a constant concatenation at runtime.

Reply:
merged all of them to a single literal with backticks!